services/user-svc/internal/organization/organization.go

```diff
@@ -685,88 +701,66 @@ func (s ServiceServer) RevokeInvitation(ctx context.Context, req *pb.RevokeInvit
 	return &pb.RevokeInvitationResponse{}, nil
 }
 
-func CreateOrganizationAndAddUser(ctx context.Context, attr models.Base, userID uuid.UUID) (*models.Organization, error) {
-	db := hwgorm.GetDB(ctx)
-
-	var organization models.Organization
-	err := db.Transaction(func(tx *gorm.DB) error {
-		organizationPtr, err := CreateOrganization(ctx, tx, attr, userID)
-		if err != nil {
-			return err
-		}
-
-		organization = *organizationPtr
+func CreateOrganizationAndAddUser(ctx context.Context, attr organization_repo.Organization, userID uuid.UUID) (*organization_repo.Organization, error) {
+	db := hwdb.GetDB()
 
-		if err := AddUserToOrganization(ctx, tx, userID, organization.ID); err != nil {
-			return err
-		}
-
-		// Make the creating user admin of the organization
-		if err := ChangeMembershipAdminStatus(ctx, tx, userID, organization.ID, true); err != nil {
-			return err
-		}
-
-		organizationCreatedEvent := &events.OrganizationCreatedEvent{
-			Id:     organization.ID.String(),
-			UserId: userID.String(),
-		}
-
-		daprClient := common.MustNewDaprGRPCClient()
-
-		if err := common.PublishMessage(ctx, daprClient, "pubsub", "ORGANIZATION_CREATED", organizationCreatedEvent); err != nil {
-			return err
-		}
+	tx, err := db.Begin(ctx)
+	if err != nil {
+		return nil, err
+	}
 
-		// TODO: Dispatch UserJoinedOrganizationEvent
+	organizationRepo := organization_repo.New(db).WithTx(tx)
 
-		return nil
+	organization, err := organizationRepo.CreateOrganization(ctx, organization_repo.CreateOrganizationParams{
+		LongName:        attr.LongName,
+		ShortName:       attr.ShortName,
+		ContactEmail:    attr.ContactEmail,
+		AvatarUrl:       attr.AvatarUrl,
+		IsPersonal:      attr.IsPersonal,
+		CreatedByUserID: userID,
 	})
-
 	if err != nil {
 		return nil, err
```

Comment:
I believe you have to rollback the transaction before we return. From what I read in the docs you could simply add a `defer tx.Rollback()` after creating the transaction, which will cause the rollback to be attempted before any return. If we commit before, the rollback won't happen. But please read up on that again :D

Reply:
Yes you are 100% right, must have missed that.